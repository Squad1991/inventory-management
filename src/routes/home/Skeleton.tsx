import { css } from '@compiled/react';
import React from 'react';
import Skeleton from 'react-loading-skeleton';
import Stack from '~src/components/common/primitive/Stack';

const HomePageSkelton: React.FC = () => {
  return (
    <Stack css={css({ gap: '.5rem', position: 'relative', lineHeight: '1.5rem' })}>
      <Skeleton width={'20%'} height={10} />
      <Skeleton
        width={'20rem'}
        inline
        style={{ marginInlineEnd: '1rem' }}
        count={20}
        height={200}
      />
    </Stack>
  );
};

export default React.memo(HomePageSkelton);
