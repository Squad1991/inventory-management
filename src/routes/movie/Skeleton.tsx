import { css } from '@compiled/react';
import React from 'react';
import Skeleton from 'react-loading-skeleton';
import Stack from '~src/components/common/primitive/Stack';

const MovieInfoPageSkelton: React.FC = () => {
  return (
    <Stack css={css({ gap: '2rem', position: 'relative', lineHeight: '1.5rem' })}>
      <Stack>
        <Skeleton enableAnimation width={'100%'} height={300} />
        <Skeleton width={'20%'} height={10} />
        <Skeleton width={'40%'} height={10} />
        <Skeleton count={3} height={10} />
      </Stack>
      <Stack>
        <Skeleton width={'20%'} height={10} />
        <Skeleton count={3} height={60} />
      </Stack>
      <Stack>
        <Skeleton width={'20%'} height={10} />
        <Skeleton
          width={'20rem'}
          inline
          style={{ marginInlineEnd: '1rem' }}
          count={3}
          height={200}
        />
      </Stack>
    </Stack>
  );
};

export default MovieInfoPageSkelton;
