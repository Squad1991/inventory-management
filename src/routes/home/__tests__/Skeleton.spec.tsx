import { render } from '@testing-library/react';
import React from 'react';
import Skeleton from '~src/routes/home/Skeleton';

describe('Skeleton', () => {
  it('should render correctly', () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });
});
