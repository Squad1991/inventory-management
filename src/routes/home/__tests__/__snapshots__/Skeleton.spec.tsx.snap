// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > should render correctly 1`] = `
<div>
  <style
    data-cmpld="true"
  >
    ._zulp1kwk{gap:.5rem}._kqswh2mm{position:relative}._vwz41k8s{line-height:1.5rem}
  </style>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
  </style>
  <div
    class="_1e0c1txw _2lx21bp4 _zulp1kwk _kqswh2mm _vwz41k8s"
  >
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
        style="width: 20%; height: 10px;"
      >
        ‌
      </span>
      <br />
    </span>
    <span
      aria-busy="true"
      aria-live="polite"
    >
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
      <span
        class="react-loading-skeleton"
        style="margin-inline-end: 1rem; width: 20rem; height: 200px;"
      >
        ‌
      </span>
    </span>
  </div>
</div>
`;
